crates/flux-fhir-analysis/src/conv.rs

```diff
@@ -193,12 +194,24 @@ pub(crate) fn conv_generics(
     })
 }
 
+fn sort_args_for_adt(genv: &GlobalEnv, def_id: impl Into<DefId>) -> List<fhir::Sort> {
+    let mut sort_args = vec![];
+    for param in &genv.tcx.generics_of(def_id.into()).params {
+        if let rustc_middle::ty::GenericParamDefKind::Type { .. } = param.kind {
+            sort_args.push(fhir::Sort::Param(param.def_id));
```

Comment:
This made me realize we are using `fhir::Sort::Param` for two unrelated reasons

1) The sort of a parameter of kind base
2) To bind the sort in a `refined_by` to a corresponding generic

I believe we should use `fhir::Sort::Var` for the second one. Let me elaborate. The annotation `refined_by(elems: Set<T>)` implicitly creates a sort `record RSet<S> { elems: Set<S> }`  this is analogous `fn foo<s>(....)` so it would be consistent to use `fhir::Sort::Var`. We will need a mapping between the `DefId` of the type parameter and the index of the sort it corresponds to so we can implement `sort_of_path` (related to https://github.com/flux-rs/flux/pull/542#discussion_r1376838230). As a nice side effect, I think we could get rid of all the `param_sort` methods in `fhir`.

Reply:
done. (I assume you meant use `subst` instead of `param_subst`? (and yes we can I removed all the `param_subst` stuff)